src/Payment/PaymentService.php

```diff
@@ -488,6 +497,14 @@ protected function processAsMolliePayment(
                 $apiKey
             )->payments->create($data);
         } catch (ApiException $e) {
+            $isMollieOutage = $this->apiHelper->isMollieOutageException($e);
+            if ($isMollieOutage) {
+                $this->logger->debug(
+                    "Creating payment object: type Order failed due to a Mollie outage, stopping process. Check Mollie status at https://status.mollie.com/. {$e->getMessage()}"
+                );
+
+                throw new ApiException(__('Payment failed due to: Mollie is out of service. Please try again later.', 'mollie-payments-for-woocommerce'));
+            }
```

Comment:
Maybe should extract to a function like `handleMollieOutage($e)` instead of duplicating.